packages/trader/src/AppV2/Utils/sort-symbols-utils.ts

```diff
@@ -0,0 +1,27 @@
+import { ActiveSymbols } from '@deriv/api-types';
+
+type MarketOrderMap = {
+    [key: string]: number;
+};
+
+const sortSymbols = (symbolsList: ActiveSymbols) => {
+    const marketSortingOrder = ['synthetic_index', 'forex', 'indices', 'cryptocurrency', 'commodities'];
+    const marketOrderMap: MarketOrderMap = marketSortingOrder.reduce(
+        (acc: MarketOrderMap, market: string, index: number) => {
+            acc[market] = index;
+            return acc;
+        },
+        {}
+    );
+
+    return symbolsList.sort((a, b) => {
+        const marketOrderA = marketOrderMap[a.market] !== undefined ? marketOrderMap[a.market] : symbolsList.length;
+        const marketOrderB = marketOrderMap[b.market] !== undefined ? marketOrderMap[b.market] : symbolsList.length;
+        if (marketOrderA !== marketOrderB) {
```

Comment:
If it's comparing non-primities, then it will be comparing links (refers). Is it what you were targeting?

Reply:
(just for my understanding, sorry for silly extra questions)

Reply:
marketOrderA returns a number actually.